x/gov/tally.go

```diff
@@ -6,35 +6,37 @@ import (
 
 // validatorGovInfo used for tallying
 type validatorGovInfo struct {
-	Address         sdk.ValAddress // address of the validator operator
-	BondedTokens    sdk.Int        // Power of a Validator
-	DelegatorShares sdk.Dec        // Total outstanding delegator shares
-	Minus           sdk.Dec        // Minus of validator, used to compute validator's voting power
-	Vote            VoteOption     // Vote of the validator
+	Address             sdk.ValAddress // address of the validator operator
+	BondedTokens        sdk.Int        // Power of a Validator
+	DelegatorShares     sdk.Dec        // Total outstanding delegator shares
+	DelegatorDeductions sdk.Dec        // Delegator deductions from validator's delegators voting independently
+	Vote                VoteOption     // Vote of the validator
 }
 
 func newValidatorGovInfo(address sdk.ValAddress, bondedTokens sdk.Int, delegatorShares,
-	minus sdk.Dec, vote VoteOption) validatorGovInfo {
+	delegatorDeductions sdk.Dec, vote VoteOption) validatorGovInfo {
 
 	return validatorGovInfo{
-		Address:         address,
-		BondedTokens:    bondedTokens,
-		DelegatorShares: delegatorShares,
-		Minus:           minus,
-		Vote:            vote,
+		Address:             address,
+		BondedTokens:        bondedTokens,
+		DelegatorShares:     delegatorShares,
+		DelegatorDeductions: delegatorDeductions,
+		Vote:                vote,
 	}
 }
 
+// TODO: Break into several smaller functions for clarity
 func tally(ctx sdk.Context, keeper Keeper, proposal Proposal) (passes bool, tallyResults TallyResult) {
-	results := make(map[VoteOption]sdk.Dec)
-	results[OptionYes] = sdk.ZeroDec()
-	results[OptionAbstain] = sdk.ZeroDec()
-	results[OptionNo] = sdk.ZeroDec()
-	results[OptionNoWithVeto] = sdk.ZeroDec()
+	results := make(map[VoteOption]sdk.Int)
```

Comment:
Have this be a map of sdk.Dec and then just do the rounding at the end when saving to TallyResult

Reply:
OK done